misk-redis/src/testFixtures/kotlin/misk/redis/testing/FakeRedis.kt

```diff
@@ -411,4 +422,96 @@ class FakeRedis @Inject constructor(
   override fun publish(channel: String, message: String) {
     throw NotImplementedError("Fake client not implemented for this operation")
   }
+
+  private fun zadd(
+    key: String,
+    score: Double,
+    member: ByteString,
+    options: Set<Redis.ZAddOptions>,
+  ): Long {
+    verifyZAddOptions(options)
+    var newFieldCount = 0L
+    var elementsChanged = 0L
+
+    if (!sortedSetKeyValueStore.containsKey(key)) {
+      sortedSetKeyValueStore[key] = Value(data = sortedMapOf(), expiryInstant = Instant.MAX)
+    }
+    val exists = sortedSetKeyValueStore[key]!!.data[member] != null
+
```

Comment:
I'd recommend handling the no-op `XX` and `NX` cases here and returning early, to help simplify the logic below:

```suggestion
    if ((options.contains(XX) && !exists) || (options.contains(NX) && exists)) {
      return 0
    }
 
```

I think this will also eliminate the bug I mention below

Reply:
This should be fixed now.